WebApplication1/Controllers/UsersController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MediatR;
+using System;
+using System.Threading.Tasks;
+using WebApplication1.Application.Queries;
+using WebApplication1.Domain;
+using WebApplication1.Application.Commands;
+using Microsoft.AspNetCore.Authorization;
+using WebApplication1.ViewModel;
+using WebApplication1.DTO;
+using WebApplication1.Application.Options;
+using AutoMapper;
+
+namespace WebApplication1.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class UsersController : ControllerBase
+    {
+        private readonly IMediator mediator;
+        private readonly IMapper mapper;
+
+        public UsersController(IMediator mediator, IMapper mapper)
+        {
+            this.mediator = mediator;
+            this.mapper = mapper;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllAsync()
+        {
+            var result = new UserGetResponse { Users = await mediator.Send(new GetUsers.Query()) };
```

Comment:
try to avoid such long constructions. It is better to rewrite similar code like bellow
```suggestion
            TypeOfResult result = await mediator.Send(new GetUsersQuery());
            var response = new UserGetResponse 
            {
                   Users = result 
            }; 
```